specification/cognitiveservices/data-plane/WebSearch/v1.0/WebSearch.json

```diff
@@ -0,0 +1,1211 @@
+            "x-ms-client-name": "AnswerCount",
+            "in": "query",
+            "description": "The number of answers that you want the response to include. The answers that Bing returns are based on ranking. For example, if Bing returns webpages, images, videos, and relatedSearches for a request and you set this parameter to two (2), the response includes webpages and images.If you included the responseFilter query parameter in the same request and set it to webpages and news, the response would include only webpages.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "cc",
+            "x-ms-client-name": "CountryCode",
+            "in": "query",
+            "description": "A 2-character country code of the country where the results come from. This API supports only the United States market. If you specify this query parameter, it must be set to us. If you set this parameter, you must also specify the Accept-Language header. Bing uses the first supported language it finds from the languages list, and combine that language with the country code that you specify to determine the market to return results for. If the languages list does not include a supported language, Bing finds the closest language and market that supports the request, or it may use an aggregated or default market for the results instead of a specified one. You should use this query parameter and the Accept-Language query parameter only if you specify multiple languages; otherwise, you should use the mkt and setLang query parameters. This parameter and the mkt query parameter are mutually exclusive—do not specify both.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "count",
+            "x-ms-client-name": "Count",
+            "in": "query",
+            "description": "The number of search results to return in the response. The default is 10 and the maximum value is 50. The actual number delivered may be less than requested.Use this parameter along with the offset parameter to page results.For example, if your user interface displays 10 search results per page, set count to 10 and offset to 0 to get the first page of results. For each subsequent page, increment offset by 10 (for example, 0, 10, 20). It is possible for multiple pages to include some overlap in results.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "freshness",
+            "x-ms-client-name": "Freshness",
+            "in": "query",
+            "description": "Filter search results by the following age values: Day—Return webpages that Bing discovered within the last 24 hours. Week—Return webpages that Bing discovered within the last 7 days. Month—Return webpages that discovered within the last 30 days. This filter applies only to webpage results and not to the other results such as news and images.",
```

Comment:
I recommend using an `enum` here:

``` json
  "enum": [
    "Day",
    "Week",
    "Month",
   ],
  "x-ms-enum" : {
    "name" : "Freshness",
    "description": "The length of time which to scope the query.",
    "values": [
      { 
        "value" : "Day",
        "description" : "Return webpages that Bing discovered within the last 24 hours."
      },
      { 
        "value" : "Week",
        "description" : "Return webpages that Bing discovered within the last 7 days."
      },
      { 
        "value" : "Month",
        "description" : "Return webpages that discovered within the last 30 days."
      }
    ]
  } 

```

Reply:
Yea, just create a Freshness enum in ApiSchemas and reference it in your endpoint definition (instead of a string)

Reply:
Updated

Reply:
If you add the `description` and `values` to the `x-ms-enum` you'll generate a better code with descriptions for each value.

``` json
   "description": "The length of time which to scope the query.",
    "values": [
      { 
        "value" : "Day",
        "description" : "Return webpages that Bing discovered within the last 24 hours."
      },
      { 
        "value" : "Week",
        "description" : "Return webpages that Bing discovered within the last 7 days."
      },
      { 
        "value" : "Month",
        "description" : "Return webpages that discovered within the last 30 days."
      }
    ]
```